misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java

```diff
@@ -1328,270 +1452,1389 @@ public ListBindingPatternNode transform(ListBindingPatternNode listBindingPatter
-                .withDocumentationLines(documentationLines)
-                .apply();
+        return super.transform(restMatchPatternNode);
     }
 
     @Override
-    public MarkdownDocumentationLineNode transform(MarkdownDocumentationLineNode markdownDocumentationLineNode) {
-        Token hashToken = formatToken(markdownDocumentationLineNode.hashToken(), 0, 0);
-        NodeList<Node> documentElements = formatNodeList(markdownDocumentationLineNode.documentElements(), 0, 0, 0, 1);
+    public FieldMatchPatternNode transform(FieldMatchPatternNode fieldMatchPatternNode) {
 
-        return markdownDocumentationLineNode.modify()
-                .withDocumentElements(documentElements)
-                .withHashToken(hashToken)
-                .apply();
+        return super.transform(fieldMatchPatternNode);
     }
 
     @Override
-    public MarkdownParameterDocumentationLineNode transform(
-            MarkdownParameterDocumentationLineNode markdownParameterDocumentationLineNode) {
-        Token hashToken = formatToken(markdownParameterDocumentationLineNode.hashToken(), 1, 0);
-        Token plusToken = formatToken(markdownParameterDocumentationLineNode.plusToken(), 1, 0);
-        Token parameterName = formatToken(markdownParameterDocumentationLineNode.parameterName(), 1, 0);
-        Token minusToken = formatToken(markdownParameterDocumentationLineNode.minusToken(), 0, 0);
-        NodeList<Node> documentElements =
-                formatNodeList(markdownParameterDocumentationLineNode.documentElements(), 0, 0, 0, 1);
+    public FunctionalMatchPatternNode transform(FunctionalMatchPatternNode functionalMatchPatternNode) {
 
-        return markdownParameterDocumentationLineNode.modify()
-                .withHashToken(hashToken)
-                .withPlusToken(plusToken)
-                .withParameterName(parameterName)
-                .withMinusToken(minusToken)
-                .withDocumentElements(documentElements)
-                .apply();
+        return super.transform(functionalMatchPatternNode);
     }
 
     @Override
-    public DocumentationReferenceNode transform(DocumentationReferenceNode documentationReferenceNode) {
-        if (documentationReferenceNode.referenceType().isPresent()) {
-            Token referenceType = formatToken(documentationReferenceNode.referenceType().get(), 1, 0);
-            documentationReferenceNode = documentationReferenceNode.modify().withReferenceType(referenceType).apply();
+    public NamedArgMatchPatternNode transform(NamedArgMatchPatternNode namedArgMatchPatternNode) {
+
+        return super.transform(namedArgMatchPatternNode);
+    }
+
+    @Override
+    public OrderByClauseNode transform(OrderByClauseNode orderByClauseNode) {
+
+        return super.transform(orderByClauseNode);
+    }
+
+    @Override
+    public OrderKeyNode transform(OrderKeyNode orderKeyNode) {
+        ExpressionNode expression = formatNode(orderKeyNode.expression(), 1, 0);
```

Comment:
I think the trailing WS/NL should change depending on the existence of the `orderDirection`